src/components/FileItem.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  JupyterLab
+} from '@jupyterlab/application'
+
+import {
+  fileStyle,
+  fileGitButtonStyle,
+  fileLabelStyle,
+  fileIconStyle,
+  changeStageButtonStyle,
+  fileButtonStyle,
+  changeStageButtonRightStyle,
+} from '../components_style/FileListStyle'
+
+import {
+  fileChangedLabelStyle,
+  fileChangedLabelBrandStyle,
+  fileChangedLabelInfoStyle,
+  selectedFileStyle,
+  selectedFileChangedLabelStyle
+} from '../components_style/GitStageStyle'
+
+
+import {
+  classes 
+} from 'typestyle/lib'
+
+import * as React from 'react'
+
+export interface IFileItemProps {
+  topRepoPath: string
+  file: any
+  app: JupyterLab
+  refresh: any
+  moveFile: Function
+  moveFileIconClass: string
+  moveFileTitle: string
+  openFile: Function
+  extractFilename: Function
+  contextMenu: Function
+  parseFileExtension: Function
+  selectedFile: number
+  updateSelectedFile: Function
+  fileIndex: number
+} 
+
+export class FileItem extends React.Component<IFileItemProps, {}> {
+  getFileChangedLabel(change: string) : string {
```

Comment:
This is a personal style preference, but if a method is pure (doesn't depend on `this`), I like hoist it out of the component class (e.g. move it out of the class and into it's own function).